transport/http/handler.go

```diff
@@ -234,15 +236,24 @@ func (h handler) createSpan(ctx context.Context, req *http.Request, treq *transp
 	return ctx, span
 }
 
+var _ transport.ResponseWriter = (*responseWriter)(nil)
+var _ transport.ResponseMetaWriter = (*responseWriter)(nil)
+
 // responseWriter adapts a http.ResponseWriter into a transport.ResponseWriter.
 type responseWriter struct {
 	w      http.ResponseWriter
 	buffer *bufferpool.Buffer
+	meta   *transport.ResponseMeta
 }
 
-func newResponseWriter(w http.ResponseWriter) *responseWriter {
-	w.Header().Set(ApplicationStatusHeader, ApplicationSuccessStatus)
-	return &responseWriter{w: w}
+func newResponseWriter(w http.ResponseWriter, headers http.Header) *responseWriter {
+	return &responseWriter{
+		w: w,
+		meta: &transport.ResponseMeta{
+			ID:      headers.Get(IDHeader),
+			Service: headers.Get(ServiceHeader),
```

Comment:
This doesn't seem right. This is reading the ID and Service that will be
written to the response from the response writer's own header map, which is
likely empty at this time.